test/src/test/java/ai/timefold/solver/test/api/score/stream/SingleConstraintAssertionTest.java

```diff
@@ -218,4 +232,151 @@ void uniquePairShouldWorkOnStringPlanningId() {
                 .rewards(1, "There should be rewards")).hasMessageContaining("There should be rewards")
                 .hasMessageContaining("Expected reward");
     }
+
+    @Test
+    void listVarUnassignedWhileAllowsUnassigned() {
+        var constraintVerifier =
+                ConstraintVerifier.build(new TestdataAllowsUnassignedListConstraintProvider(),
+                        TestdataAllowsUnassignedValuesListSolution.class,
+                        TestdataAllowsUnassignedValuesListEntity.class,
+                        TestdataAllowsUnassignedValuesListValue.class);
+
+        var value1 = new TestdataAllowsUnassignedValuesListValue("v1");
+        var value2 = new TestdataAllowsUnassignedValuesListValue("v2");
+        var entity = new TestdataAllowsUnassignedValuesListEntity("eA");
+        entity.setValueList(Collections.singletonList(value1));
+        value1.setIndex(0);
+        value1.setEntity(entity);
+
+        assertThatCode(() -> constraintVerifier
```

Comment:
I don't think you tested the case where the assertion fails for SingleConstraintAssertionTest (but that probably covered in other tests).